File: app/controllers/api/reports_controller.rb
class Api::ReportsController < Api::ApplicationController
  def create
    server = Server.where(token: request.headers['SSSM-Token']).first rescue nil
    unless server.nil?
      report = server.reports.new(report_params)
      report.ip = request.remote_ip
      report.save
    end
    render json: ""
  end

  private

  def report_params
    params.require(:report).permit!
  end
end

File: src/main/java/app/sssm/controllers/api/ReportsController.java
package app.sssm.controllers.api;

import app.sssm.domains.Server;
import app.sssm.services.ReportService;
import app.sssm.services.ServerService;
import app.sssm.domains.Report;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@RestController
public class ReportsController {

    @Autowired
    ServerService serverService;

    @Autowired
    ReportService reportService;

    @PostMapping("/api/reports")
    public void create(@RequestBody Report report,
                       @RequestHeader("Server-Token") String serverToken,
                       HttpServletResponse response,
                       HttpServletRequest request) {
        Server server = serverService.getByToken(serverToken);
        if (server != null) {
            reportService.save(report, server, request);
        } else {
            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
        }
    }

}
